zoombot/zoombot/api.go

```diff
@@ -156,3 +158,24 @@ func CreateMeeting(client *http.Client, userID string, request *CreateMeetingReq
 
 	return &meeting, nil
 }
+
+type ErrorResponse struct {
+	Code    int    `json:"code"`
+	Message string `json:"message"`
+}
+
+var MaxMeetingsError = errors.New("a maximum of 100 meetings can be created/updated for a single user in one day")
+
+func parseError(statusCode int, data []byte) error {
+	var errorResponse ErrorResponse
+	err := json.Unmarshal(data, &errorResponse)
+	if err != nil {
+		return fmt.Errorf("statusCode: %d, error: %s", statusCode, data)
+	}
+	switch errorResponse.Message {
+	case "A maximum of 100 meetings can be created/updated for a single user in one day.":
```

Comment:
using the status code seems safter

Reply:
I set up this `parseError` func so it could be used with other api methods (I'll need to add some for the compliance stuff). Maybe that was a bad idea? If we stuck with one `parseError` func using the statusCode wouldn't be enough to classify the error (there are different reasons why Zoom might rate limit us with a code 429).